[CI] WriteLoadForecasterIT testWriteLoadForecastGetsPopulatedDuringRollovers failing

**Build Scans:**
- [elasticsearch-intake #27949 / part2](https://gradle-enterprise.elastic.co/s/ke5zj4dvuwcjo)
- [elasticsearch-intake #27945 / part2](https://gradle-enterprise.elastic.co/s/rggyulzd5wqiw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:write-load-forecaster:internalClusterTest" --tests "org.elasticsearch.xpack.writeloadforecaster.WriteLoadForecasterIT.testWriteLoadForecastGetsPopulatedDuringRollovers" -Dtests.seed=26B7C245B5A2612A -Dtests.locale=syr-IQ -Dtests.timezone=America/Nuuk -Druntime.java=24
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.writeloadforecaster.WriteLoadForecasterIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testWriteLoadForecastGetsPopulatedDuringRollovers'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: timed out waiting for green state
```

**Issue Reasons:**
- [9.1] 2 consecutive failures in test testWriteLoadForecastGetsPopulatedDuringRollovers
- [9.1] 2 consecutive failures in step part2
- [9.1] 2 consecutive failures in pipeline elasticsearch-intake
- [9.1] 2 failures in test testWriteLoadForecastGetsPopulatedDuringRollovers (2.3% fail rate in 87 executions)
- [9.1] 2 failures in step part2 (28.6% fail rate in 7 executions)
- [9.1] 2 failures in pipeline elasticsearch-intake (28.6% fail rate in 7 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/write-load-forecaster/src/internalClusterTest/java/org/elasticsearch/xpack/writeloadforecaster/WriteLoadForecasterIT.java
+++ b/x-pack/plugin/write-load-forecaster/src/internalClusterTest/java/org/elasticsearch/xpack/writeloadforecaster/WriteLoadForecasterIT.java
@@ -145,6 +145,8 @@ public class WriteLoadForecasterIT extends ESIntegTestCase {
     private void setUpDataStreamWriteDocsAndRollover(String dataStreamName, Settings extraIndexTemplateSettings) throws Exception {
         final int numberOfShards = randomIntBetween(1, 5);
         final int numberOfReplicas = randomIntBetween(0, 1);
+        internalCluster().ensureAtLeastNumDataNodes(numberOfReplicas + 1);
+
         final Settings indexSettings = Settings.builder()
             .put(extraIndexTemplateSettings)
             .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, numberOfShards)
@@ -194,6 +196,7 @@ public class WriteLoadForecasterIT extends ESIntegTestCase {
 
             assertAcked(indicesAdmin().rolloverIndex(new RolloverRequest(dataStreamName, null)).actionGet());
         }
+        ensureGreen();
     }
 
     static void indexDocs(String dataStream, int numDocs) {

